overlord/ifacestate/ifacemgr.go

```diff
@@ -78,6 +75,9 @@ func (m *InterfaceManager) Init(s *state.State) error {
 }
 
 func (m *InterfaceManager) doConnect(task *state.Task) error {
+	task.State().Lock()
+	defer task.State().Unlock()
```

Comment:
Hmm, something has to lock though (in current codebase) so that Get()s can reach the data. Here it doesn't matter as Repository.Connect() has private locks but in the grand scheme of things I'd like to see how/where we expect concurrency and where serialization.